modules/custom/ymca_sync/src/SyncerRunner.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Drupal\ymca_sync;
+
+use Drupal\Core\Config\ConfigFactory;
+use Drupal\Core\Lock\LockBackendInterface;
+use Drupal\Core\Logger\LoggerChannelInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class SyncerRunner.
+ *
+ * @package Drupal\ymca_sync.
+ */
+class SyncerRunner {
+
+  /**
+   * The used lock backend instance.
+   *
+   * @var \Drupal\Core\Lock\LockBackendInterface
+   */
+  protected $lock;
+
+  /**
+   * LoggerChannelInterface.
+   *
+   * @var Drupal\Core\Logger\LoggerChannelInterface
+   *   LoggerChannelInterface.
+   */
+  protected $logger;
+
+  /**
+   * ConfigFactory.
+   *
+   * @var \Drupal\Core\Config\ConfigFactory
+   */
+  protected $configFactory;
+
+  /**
+   * YMCA sync config.
+   *
+   * @var \Drupal\Core\Config\Config|\Drupal\Core\Config\ImmutableConfig
+   */
+  protected $config;
+
+
+  /**
+   * ContainerInterface.
+   *
+   * @var \Symfony\Component\DependencyInjection\ContainerInterface
+   *   ContainerInterface.
+   */
+  protected $container;
+
+  /**
+   * Implement construct method.
+   *
+   * @param \Drupal\Core\Lock\LockBackendInterface $lock
+   *   Lock.
+   * @param \Drupal\Core\Logger\LoggerChannelInterface $logger
+   *   LoggerChannelInterface.
+   * @param \Drupal\Core\Config\ConfigFactory $configFactory
+   *   ConfigFactory.
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   ContainerInterface.
+   */
+  public function __construct(LockBackendInterface $lock, LoggerChannelInterface $logger, ConfigFactory $configFactory, ContainerInterface $container) {
+    $this->lock = $lock;
+    $this->logger = $logger;
+    $this->configFactory = $configFactory;
+    $this->config = $this->configFactory->get('ymca_sync.settings');
+    $this->container = $container;
+  }
+
+  /**
+   * Run sync.
+   *
+   * @param string $name
+   *   Service name.
+   * @param string $method
+   *   Method name.
+   * @param array $options
+   *   Options.
+   *
+   * @throws \Exception
+   */
+  public function run($name, $method, array $options = []) {
+    $active = $this->config->get('active_syncers');
+
+    // Run only active syncers.
+    if (!in_array($name, $active)) {
+      $msg = 'Syncer "%name" is not activate or found. Exit. ';
```

Comment:
English: '...is not activated or found...' OR '...is not active or found...'
Another thing ("vkusovshina") - I would prefer the blank symbol to be added in the next line, not hang here at the end.